src/scheduled_tasks.rs

```diff
@@ -23,33 +31,35 @@ pub fn setup(db_url: String, user_agent: String) -> Result<(), LemmyError> {
   // Setup the connections
   let mut scheduler = Scheduler::new();
 
-  let mut conn = PgConnection::establish(&db_url).expect("could not establish connection");
-
+  let mut conn_1 = PgConnection::establish(&db_url).expect("could not establish connection");
   let mut conn_2 = PgConnection::establish(&db_url).expect("could not establish connection");
+  let mut conn_3 = PgConnection::establish(&db_url).expect("could not establish connection");
+  let mut conn_4 = PgConnection::establish(&db_url).expect("could not establish connection");
 
-  active_counts(&mut conn);
-  update_banned_when_expired(&mut conn);
+  // Run on startup
+  active_counts(&mut conn_1);
+  update_hot_ranks(&mut conn_1, false);
+  update_banned_when_expired(&mut conn_1);
+  clear_old_activities(&mut conn_1);
 
-  // On startup, reindex the tables non-concurrently
-  // TODO remove this for now, since it slows down startup a lot on lemmy.ml
-  reindex_aggregates_tables(&mut conn, true);
+  // Update active counts every hour
   scheduler.every(CTimeUnits::hour(1)).run(move || {
-    let conn = &mut PgConnection::establish(&db_url)
-      .unwrap_or_else(|_| panic!("Error connecting to {db_url}"));
-    active_counts(conn);
-    update_banned_when_expired(conn);
-    reindex_aggregates_tables(conn, true);
-    drop_ccnew_indexes(conn);
+    active_counts(&mut conn_1);
+    update_banned_when_expired(&mut conn_1);
+  });
+
+  // Update hot ranks every 5 minutes
+  scheduler.every(CTimeUnits::minutes(5)).run(move || {
+    update_hot_ranks(&mut conn_2, true);
   });
 
-  clear_old_activities(&mut conn);
+  // Clear old activities every week
   scheduler.every(CTimeUnits::weeks(1)).run(move || {
-    clear_old_activities(&mut conn);
+    clear_old_activities(&mut conn_3);
   });
 
-  update_instance_software(&mut conn_2, &user_agent);
```

Comment:
I had to remove this one. While these don't block the main server pool, they do block the schedulers from starting, and the hot_ranks do really need to start right away. Every other one took < 1 minute, but updating the instance stats seemed like it was going to take many hours. We might also want to change this to weekly rather than daily.

Reply:
Makes sense, the instance list is very long and its not necessary to update on startup. In the future we can make it smarter to not connect to instances which have been unreachable previously.

Alternatively it could run separate jobs to update like 5 instances at a time, to spread out the load.